File: scratch.py
import pandas as pd

f1 = r'C:\Users\priya\Downloads\barshi.xlsx'
f2 = r'C:\Users\priya\Downloads\9afe08cb1b664f27b46f248efa3f82c5.xlsx'

print('--- barshi.xlsx ---')
try:
    df1 = pd.read_excel(f1)
    print('Columns:', df1.columns.tolist())
    mask = df1.apply(lambda row: row.astype(str).str.contains('19004214').any(), axis=1)
    print(df1[mask].to_string())
except Exception as e:
    print('Error:', e)

print('\n--- closing stock ---')
try:
    df2 = pd.read_excel(f2)
    print('Columns:', df2.columns.tolist())
    mask2 = df2.apply(lambda row: row.astype(str).str.contains('19004214').any(), axis=1)
    print(df2[mask2].to_string())
except Exception as e:
    print('Error:', e)

File: scratch.mjs
import xlsx from 'xlsx';

const file = 'C:\\Users\\priya\\Downloads\\01f69f992bb74877a5ec70fbd4f009a3.xlsx';

try {
    const wb = xlsx.readFile(file);
    const sheetName = wb.SheetNames[0];
    const sheet = wb.Sheets[sheetName];
    const data = xlsx.utils.sheet_to_json(sheet, { header: 1 });
    console.log('Rows:', data.length);
    // Find the header row (assume it's the one with 'BRANCH NAME' or 'CLOSING STOCK')
    let headerIdx = -1;
    for (let i=0; i<Math.min(20, data.length); i++) {
        if (data[i] && data[i].some(c => String(c).toUpperCase().includes('BRANCH NAME'))) {
            headerIdx = i;
            break;
        }
    }
    console.log('Header Index:', headerIdx);
    if (headerIdx !== -1) {
        console.log('Headers:', data[headerIdx]);
        console.log('Sample Data 1:', data[headerIdx + 1]);
        console.log('Sample Data 2:', data[headerIdx + 2]);
    }
} catch (e) {
    console.error('Error reading file:', e);
}
